test.sh.in

```diff
@@ -358,6 +367,126 @@ fetchOVALog(){
     popd
 }
 
+prepareDockerImage(){
+    pushd $WORKSPACE
+    echo "[Info] Downloading rackhd_pipeline_docker image"
+
+    # -c  resume getting a partially-downloaded file.
+    # -nv reduce the verbose output
+    # -t 5  the retry counter
+    wget -c -t 5 -nv --no-check-certificate http://rackhdci.lss.emc.com/job/Archiving/job/BuildDockerImage/lastSuccessfulBuild/artifact/rackhd_pipeline_docker.tar
+
+    if [ $? -ne 0 ]; then
+        echo "[Error]: wget download failed: ${remote_file}"
+        exit 2
+    else
+        echo "[Info] wget download successfully ${remote_file}"
+    fi
+    echo $SUDO_PASSWORD |sudo -S docker load -i rackhd_pipeline_docker.tar
+    popd
+}
+
+dockerUp(){
+    prepareDockerImage
+    cp -r ${WORKSPACE}/build-deps ${WORKSPACE}/build-config/jobs/pr_gate/docker
+    pushd ${WORKSPACE}/build-config/jobs/pr_gate/docker
+    #cp -r ${WORKSPACE}/build-config/jobs/pr_gate/docker/* .
+    echo $SUDO_PASSWORD |sudo -S docker build -t my/test .
+    echo $SUDO_PASSWORD |sudo -S docker run --net=host -d -t my/test
+    popd
+}
+
+setupTestsConfig(){
+    if [ "$RUN_FIT_TEST" == true ] ; then
+        cp ${WORKSPACE}/build-config/jobs/pr_gate/test_config/FIT/*.* ${WORKSPACE}/RackHD/test/config
+    fi
+    if [ "$RUN_CIT_TEST" == true ] ; then
+        cp ${WORKSPACE}/build-config/jobs/pr_gate/test_config/CIT/config.ini ${WORKSPACE}/RackHD/test/config/
+    fi
+    pushd ${WORKSPACE}/RackHD/test/config
+    sed -i "s/RACKHD_SSH_USER = vagrant/RACKHD_SSH_USER = ${SUDO_USER}/g" config.ini
+    sed -i "s/RACKHD_SSH_PASSWORD = vagrant/RACKHD_SSH_PASSWORD = $SUDO_PASSWORD/g" config.ini
+    sed -i "s/\"username\": \"vagrant\"/\"username\": \"${SUDO_USER}\"/g" credentials_default.json
+    sed -i "s/\"password\": \"vagrant\"/\"password\": \"$SUDO_PASSWORD\"/g" credentials_default.json
+    popd
+}
+
+runTestsForDocker() {
+    set +e
+    netstat -ntlp
+    setupTestsConfig
+    if [ "$RUN_FIT_TEST" == true ] ; then
+        fitSmokeTest
+    fi
+    if [ "$RUN_CIT_TEST" == true ] ; then
+        citSmokeTest
+    fi
+    set -e
+}
+
+collectTestReport()
+{
+    pushd ${WORKSPACE}/RackHD/test
+    mkdir -p ${WORKSPACE}/xunit-reports
+    cp *.xml ${WORKSPACE}/xunit-reports
+    popd
+}
+
+fitSmokeTest()
+{
+    set +e
+    echo "########### Run FIT Stack Init #############"
+    pushd ${WORKSPACE}/RackHD/test
+    #TODO Parameterize FIT args
+    echo ${TEST_STACK}
+    tstack="${TEST_STACK}"
+    read hwarray<<<"${EXTRA_HW}"
+    for item in $hwarray; do
+        if [ "${item}" == "ucs" ] && [ "${TEST_STACK}" == "-stack vagrant" ]; then
+            tstack="-stack vagrant_ucs"
```

Comment:
ova post test is now using 'stack ova' from rackhd/test/config/stack-config.json.
 from run #225:  20:09:39 20:09:39 [ova FIT] + python run_tests.py -test deploy/rackhd_stack_init.py -stack ova -xunit

docker post test should be updated to use stack docker.  (We have been working on that testing on test jenkins).